streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamKTableJoinProcessor.java

```diff
@@ -42,24 +42,24 @@
 import static org.apache.kafka.streams.processor.internals.metrics.TaskMetrics.droppedRecordsSensor;
 import static org.apache.kafka.streams.state.ValueAndTimestamp.getValueOrNull;
 
-class KStreamKTableJoinProcessor<K1, K2, V1, V2, VOut> extends ContextualProcessor<K1, V1, K1, VOut> {
+class KStreamKTableJoinProcessor<KStream, VStream, KTable, VTable, VOut> extends ContextualProcessor<KStream, VStream, KStream, VOut> {
```

Comment:
Flipping type order -- no splash radius -- change limited to this line only, but diff is weird.

Effectively renaming `K1 -> KStream`, `V1 -> VStream`, `K2 -> KTable`, and `V2 -> VTable`.

Reply:
@bbejeck Mentioned on some other PR, that `KStream` (and `KTable`) is a confusing generic type var, as we also have interfaces `KStream` and `KTable`.

Might be good to rename this differently, but I am not sure how. Ideas?

Reply:
`KLeft` and `KRight`?

Reply:
Maybe.

I was thinking about `StreamKey`/`StreamValue`/`TableKey`/`TableValue` to highlight that it's two different "objects". Left/Right would not express this explicitly.